### PR TITLE
Add log_key_prefix to WandbLogger to control the prefix for global_step

### PR DESCRIPTION
**Pull request by pupeno**
## What does this PR do?

Contributes to #21782

I made a quick video overview that might save you some time during review: https://youtu.be/AVeU2xPCRkg

Adds `log_key_prefix` to `WandbLogger` to let users control the W&B-generated step axis key,
and changes the default from the hardcoded `"trainer/global_step"` to bare `"global_step"`.

**What changes:**

- New `log_key_prefix` parameter controls the key used in `define_metric` and `experiment.log`
  for the W&B step axis. Default `None`/`""` gives bare `global_step`.
- The existing `prefix=` argument is unchanged — it continues to apply only to pass-through
  metric keys from `self.log(...)`.

**Breaking change:** `WandbLogger` previously hardcoded `"trainer/global_step"` as the step axis
key. The new default is bare `"global_step"`. Pass `log_key_prefix="trainer/"` to restore the
previous behaviour.

```python
# before: always used "trainer/global_step"
logger = WandbLogger(...)

# after: bare "global_step" by default
logger = WandbLogger(...)

# restore previous behaviour
logger = WandbLogger(log_key_prefix="trainer/")

# align with a Trainer(log_key_prefix="trainer/") setup
logger = WandbLogger(log_key_prefix="trainer/")
```

Sample code can be found on https://github.com/pupeno/pytorch-lightning-workspace/tree/main/21785_wandb_log_key_prefix

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

### CODE REVIEW

**Comment by pupeno** [comment]
Merged master in, and there's some sample code on https://github.com/pupeno/pytorch-lightning-workspace/blob/main/scratch/21785/experiment.py

Anything else I can do here?
